pkg/cmd/experimental/syncgroups/sync-groups.go

```diff
@@ -0,0 +1,308 @@
+package syncgroups
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api/latest"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util"
+	kerrs "k8s.io/kubernetes/pkg/util/errors"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/schema1"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+	serverapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	SyncGroupsRecommendedName = "sync-groups"
+
+	syncGroupsLong = `
+Sync OpenShift Groups with records from an external provider.
+
+In order to sync OpenShift Group records with those from an external provider, determine which Groups you wish
+to sync and where their records live. For instance, all or some groups may be selected from the current Groups
+stored in OpenShift, or similarly all or some groups may be selected from those stored on an LDAP server. The
+path to a sync configuration file is required in order to describe how data is requested from the external 
+record store and migrated to OpenShift records.`
+	syncGroupsExamples = `  // Sync all groups from an LDAP server
+  $ %[1]s ldap all --sync-config=/path/to/ldap-sync-config.yaml
+
+  // Sync specific groups specified in a whitelist file with an LDAP server 
+  $ %[1]s ldap whitelist --whitelist-source=/path/to/whitelist.txt --sync-config=/path/to/sync-config.yaml
```

Comment:
While these seem like sane defaults I feel like they greatly hinder UX as now each of the four choices is a "special case." In my current scheme, sure, `whitelist-source` may be redundant, but at least each and ever incarnation of `sync-groups` accepts the same form - `source scope args`. @liggitt @smarterclayton does that matter at all?